eth_createAccessList - Closes #4081 (#4332)
ethereum/web3.js: docs/web3-eth-contract.rst

    myContract.methods.myMethod([param1[, param2[, ...]]])

> Creates a transaction object for that method, which then can be :ref:`called <contract-call>`, :ref:`send <contract-send>`, :ref:`estimated  <contract-estimateGas>`, :ref:`estimated  <contract-createAccessList>` , or :ref:`ABI encoded <contract-encodeABI>`.

Author: ```suggestion
Creates a transaction object for that method, which then can be :ref:`called <contract-call>`, :ref:`send <contract-send>`, :ref:`estimated  <contract-estimateGas>`, :ref:`createAccessList  <contract-createAccessList>` , or :ref:`ABI encoded <contract-encodeABI>`.
```

- ``Function`` - :ref:`send <contract-send>`: Will send a transaction to the smart contract and execute its method (Can alter the smart contract state).
- ``Function`` - :ref:`estimateGas <contract-estimateGas>`: Will estimate the gas used when the method would be executed on chain. Note: You must specify a ``from`` address otherwise you may experience odd behavior.
- ``Function`` - :ref:`encodeABI <contract-encodeABI>`: Encodes the ABI for this method. This can be send using a transaction, call the method or passing into another smart contracts method as argument.

> - ``Function`` - :ref:`createAccessList <contract-createAccessList>`: Returns an EIP-2930 access list for specified contract method Note: You must specify a ``from`` address and possible ``gas``

Author: ```suggestion
- ``Function`` - :ref:`createAccessList <contract-createAccessList>`: Returns an EIP-2930 access list for specified contract method Note: You must specify a ``from`` address and ``gas`` if it's not specified in ``options`` when instantiating parent contract object (e.g. ``new web3.eth.Contract(jsonInterface[, address][, options])``).
```

    > '0x58cf5f1000000000000000000000000000000000000000000000000000000000000007B'

------------------------------------------------------------------------------

.. _contract-createAccessList:

methods.myMethod.createAccessList
=====================

.. code-block:: javascript

    myContract.methods.myMethod([param1[, param2[, ...]]]).createAccessList(options, blockHashOrBlockNumber [, callback])

Will call to create an access list a method execution will access when executed in the EVM.
Note: Currently `eth_createAccessList` seems to only be supported by Geth.
Note: You must specify a ``from`` address and ``gas`` if it's not specified in ``options`` when instantiating parent contract object (e.g. ``new web3.eth.Contract(jsonInterface[, address][, options])``).

----------
Parameters
----------

> 1. ``options`` - ``Object`` (optional): The options used for calling.

Author: ```suggestion
1. ``options`` - ``Object``: The options used for calling.
```

>     * ``from`` - ``String`` (optional): The address the call "transaction" should be made from.

Author: ```suggestion
    * ``from`` - ``String``: The address the call "transaction" should be made from.
```

    * ``gas`` - ``Number`` (optional): The maximum gas provided for this call "transaction" (gas limit). Setting a specific value helps to detect out of gas errors. Access list response will return amount of gas used.

> 2. ``block`` - ``String|Number|BN|BigNumber`` - The block number or hash. Or the string ``"earliest"``, ``"latest"`` or ``"pending"`` as in the :ref:`default block parameter <eth-defaultblock>`.

Author: ```suggestion
2. ``block`` - ``String|Number|BN|BigNumber`` (optional): The block number or hash. Or the string ``"earliest"``, ``"latest"`` or ``"pending"`` as in the :ref:`default block parameter <eth-defaultblock>`.
```
